scylla-cql/src/frame/response/mod.rs

```diff
@@ -88,9 +88,11 @@ impl Response {
         Ok(response)
     }
 
-    pub fn into_non_error_response(self) -> Result<NonErrorResponse, QueryError> {
+    pub fn into_non_error_response(self) -> Result<NonErrorResponse, UserRequestError> {
         Ok(match self {
-            Response::Error(err) => return Err(QueryError::from(err)),
+            Response::Error(error::Error { error, reason }) => {
+                return Err(UserRequestError::DbError(error, reason))
+            }
```

Comment:
This is a nontrivial change. Is the `<From<_> for QueryError>::from` implementation equivalent to the introduced code?

Reply:
Yes, the `From` impl:
```rust
impl From<Error> for QueryError {
    fn from(error: Error) -> QueryError {
        QueryError::DbError(error.error, error.reason)
    }
}
```

Actually, we can now remove this `From` implementation (or replace `QueryError` with `UserRequestError` there). It is not used anywhere after this PR.

Reply:
> Actually, we can now remove this `From` implementation (or replace `QueryError` with `UserRequestError` there). It is not used anywhere after this PR.

Definitely, let's clean up legacy conversion implementations.

Reply:
Nvm, this is one is actually used in one place. However, the `From<std::io::Error> for QueryError` can be removed.